third_party/2and3/mypy_extensions.pyi

```diff
@@ -1,6 +1,11 @@
-from typing import Dict, Type, TypeVar
+from typing import Dict, Type, TypeVar, Union
 
 T = TypeVar('T')
 
 
 def TypedDict(typename: str, fields: Dict[str, Type[T]]) -> Type[dict]: ...
+
+# Return type that indicates a function does not return.
+# This type is equivalent to the None type, but the no-op Union is necessary to
+# distinguish the None type from the None value.
+NoReturn = Union[None]
```

Comment:
I am not sure I understand this. ``NoReturn`` is different from the ``NoneType``, the latter is a type with one value, while the former is a type with no values, a.k.a. "error type" or "bottom type". Maybe it would be better to write ``class NoReturn: ...``?

@gvanrossum What do you think about this?